src/python/pants/backend/python/rules/coverage_integration_test.py

```diff
@@ -143,42 +145,92 @@ def test_coverage(self) -> None:
                 -----------------------------------------------------------------------------------------------
-                TOTAL                                                             8      0      0      0   100%
+                TOTAL                                                            19      2      0      0    89%
                 """
             )
             in result.stderr_data
         )
+
+    def _assert_html_report(self, result: PantsResult, build_root: str) -> None:
+        assert "Wrote html coverage report to `dist/coverage/python`" in result.stderr_data
+        html_cov_dir = Path(build_root, "dist", "coverage", "python", "htmlcov")
+        assert html_cov_dir.exists() is True
+        assert (html_cov_dir / "index.html").exists() is True
+
+    def _assert_xml_report(self, result: PantsResult, build_root: str) -> None:
+        assert "Wrote xml coverage report to `dist/coverage/python`" in result.stderr_data
+        xml_coverage = Path(build_root) / "dist" / "coverage" / "python" / "coverage.xml"
+        assert xml_coverage.exists() is True
+        root = ElementTree.parse(xml_coverage.as_posix()).getroot()
+        assert root.tag == "coverage"
+        assert "version" in root.attrib
+        assert len(root) == 2
+        assert root[0].tag == "sources"
+        assert root[1].tag == "packages"
+
+    def test_coverage_xml(self) -> None:
+        build_root = get_buildroot()
+        with temporary_dir(root_dir=build_root) as tmpdir:
+            tmpdir_relative = self._prepare_sources(tmpdir, build_root)
+            result = self._run_tests(tmpdir_relative, "--coverage-py-report=xml")
+            self._assert_xml_report(result, build_root)
+
+    def test_coverage_html(self) -> None:
+        build_root = get_buildroot()
+        with temporary_dir(root_dir=build_root) as tmpdir:
+            tmpdir_relative = self._prepare_sources(tmpdir, build_root)
+            result = self._run_tests(tmpdir_relative, "--coverage-py-report=html")
+            self._assert_html_report(result, build_root)
+
+    def test_coverage_xml_and_console(self) -> None:
+        build_root = get_buildroot()
+        with temporary_dir(root_dir=build_root) as tmpdir:
+            tmpdir_relative = self._prepare_sources(tmpdir, build_root)
+            result = self._run_tests(tmpdir_relative, "--coverage-py-report=['xml','console']")
+            self._assert_xml_report(result, build_root)
+            self._assert_console_report(result, tmpdir_relative)
+
+    def test_coverage_html_and_xml(self) -> None:
+        build_root = get_buildroot()
+        with temporary_dir(root_dir=build_root) as tmpdir:
+            tmpdir_relative = self._prepare_sources(tmpdir, build_root)
+            result = self._run_tests(tmpdir_relative, "--coverage-py-report=['html','xml']")
+            self._assert_xml_report(result, build_root)
+            self._assert_html_report(result, build_root)
```

Comment:
While this is really comprehensive, I don't recommend having so many tests. There is a real cost to each of these integration tests (about 5-10 seconds each). This adds up. We want to be careful to avoid returning back to the state of Pants 1.x. 

I think it would be comprehensive enough to have a single new test that does both HTML and XML in one single run. It'd be fine to include the console in that, too, but fine to leave off.